CHANGELOG.md

```diff
@@ -1,3 +1,349 @@
+   LDK peers, and feature signaling may change in a future version of LDK,
+   breaking compatibility. This is negotiated automatically for
+   manually-accepted inbound channels and negotiated for outbound channels based
+   on `ChannelHandshakeConfig::negotiate_anchors_zero_fee_htlc_tx`.
+
+ * `Event::BumpTransaction` is now always generated even if the transaction has
+   sufficient fee. This allows you to manage transaction broadcasting more
+   granularly for anchor channels (#4001).
+ * The local key which receives non-HTLC-encumbered funds when the counterparty
+   force-closes a channel is now one of a static list of 1000 keys when using
+   `KeysManager` if `v2_remote_key_derivation` is set or after splicing (#4117).
+ * LSPS5 support was added, providing a push notification API for LSPS clients.
+ * Client-trusts-LSP is now supported on LSPS2 service (#3838).
+ * `LSPS2ClientEvent` now has events for failure events (#3804).
+ * `LSPS2ServiceHandler::channel_open_abandoned` was added (#3712).
+ * `Event::PendingHTLCsForwardable` has been replaced with regular calls to
+   `process_pending_htlc_forwards` in the background processor when
+   `ChannelManager::needs_pending_htlc_processing` is true (#3891, #3955).
+ * `Event::HTLCHandlingFailed`s now include a`LocalHTLCFailureReason`, providing
+   much more granular reasons for HTLCs having been failed (#3744, etc).
+ * `Event::HTLCHandlingFailed` is now generated any time forwarding an HTLC
+   fails, i.e. including cases where the HTLC onion is invalid (#2933).
+ * `OffersMessageFlow` was introduced to make it easier to implement most of the
+   BOLT 12 flows without using a `ChannelManager` (#3639).
+ * `ChannelManager::pay_for_bolt11_invoice` was added (#3617).
+ * `ChannelManager::pay_for_offer_from_human_readable_name` has been deprecated
+   in favor of the `bitcoin-payment-instructions` and
+   `ChannelManager::pay_for_offer_from_hrn`. Language bindings users may still
+   wish to use the original (#3903, #4083).
+ * `lightning::util::anchor_channel_reserves` was added to assist in estimating
+   on-chain fund requirements for anchor channel closures (#3487).
+ * Using both asynchronous and synchronous `ChannelMonitor[Update]` persistence
+   on the same `ChannelManager` will now panic. This never functioned correctly
+   and is now detected to prevent issues (#3737).
+ * LDK can now validate if HTLC errors have been tampered with (once nodes
+   upgrade). It also reports and logs the amount of time an HTLC was held so
+   that (as nodes upgrade) slow nodes can be found (#2256, #3801, other fixes).
+ * Repeated `Listen::block_disconnected` calls for each disconnected block in a
+   reorg have been replaced with a single `blocks_disconnected` call with the
+   fork point block (i.e. the highest block on both sides of the reorg, #3876).
+ * `lightning::routing::scoring::CombinedScorer` was added to better support
```

Comment:
Could mention this allows to actually merge datasets?

Reply:
Maybe that's too much detail? This changelog is pretty high level.